lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -622,6 +626,62 @@ dummySkeleton inputCount outputs = SelectionSkeleton
         TokenMap.empty
     }
 
+_evaluateMinimumFee
+    :: Cardano.ProtocolParameters
+    -> SealedTx
+    -> Coin
+_evaluateMinimumFee pp tx =
+    fromCardanoLovelace minFee
+  where
+    minFee = case getSealedTxBody tx of
+        InAnyCardanoEra ShelleyEra txbody ->
+            let (Cardano.ShelleyTxBody _ ledgertxbody _ _ _ _) = txbody
+                certNum = length $ Shelley._certs ledgertxbody
+            in Cardano.evaluateTransactionFee @Cardano.ShelleyEra pp txbody (witsNum txbody certNum) 0
+        InAnyCardanoEra AllegraEra txbody ->
+            let (Cardano.ShelleyTxBody _ ledgertxbody _ _ _ _) = txbody
+                certNum = length $ ShelleyMA.certs' ledgertxbody
+            in Cardano.evaluateTransactionFee @Cardano.AllegraEra pp txbody (witsNum txbody certNum) 0
+        InAnyCardanoEra MaryEra txbody ->
+            let (Cardano.ShelleyTxBody _ ledgertxbody _ _ _ _) = txbody
+                certNum = length $ ShelleyMA.certs' ledgertxbody
+            in Cardano.evaluateTransactionFee @Cardano.MaryEra pp txbody (witsNum txbody certNum) 0
+        InAnyCardanoEra AlonzoEra txbody ->
+            let (Cardano.ShelleyTxBody _ ledgertxbody _ _ _ _) = txbody
+                certNum = length $ Alonzo.txcerts ledgertxbody
```

Comment:
I think these `certNum` could be a slight overestimate, as not all certs require witnesses (KeyReg being the exception). This should affect the result of `_evaluateMinimumFee` when Daedalus users delegate for the first time.

Another, more theoretical case would also be where all the certs require just a single key witness.

Not a big deal, but could add a comment about it.

Reply:
6faea90